### PR TITLE
add integration tests from flask endpoint to AI API call

### PR DESCRIPTION
**Pull request by davidsbailey**
Finishes https://codedotorg.atlassian.net/browse/AITT-497. This PR adds integration testing to aiproxy, which differs from the existing unit tests because it attempts to cover the entire codepath for the assessment route (which we use to evaluate student work in production), stubbing out only the AI API request.

### CODE REVIEW

File: tests/routes/test_assessment_routes.py [hunk 1]
```diff
@@ -1,97 +1,175 @@
...
+        get_bedrock_client_mock.assert_called_once()
+
+
+class TestPostAssessmentUnitTests:
+    """ Tests POST to '/assessment' to start an assessment.
+
+    These tests stub out the Label class to test the API endpoint in isolation.
+    """
+
+    def test_should_return_400_when_no_code(self, client, randomstring):
```
Comment (davidsbailey) [inline tests/routes/test_assessment_routes.py:104]: no change to this test or the ones below, except for `test_should_return_413_on_request_too_large_error` which is moved to above. apologies for the weird diffs.

File: lib/assessment/label.py [hunk 2]
```diff
@@ -144,7 +144,7 @@ def bedrock_anthropic_label_student_work(self, prompt, rubric, student_code, stu
         if '\\"Stretch\\"' in response_body["content"][0]["text"]:
             response_body["content"][0]["text"] = response_body["content"][0]["text"].replace('\\"Stretch\\"', "“Stretch”")   
 
-        data = self.get_response_data_if_valid(response_body, rubric, student_id, response_type='json')
+        data = self.get_response_data_if_valid(response_body, rubric, student_id, response_type='json', reraise=True)
```
Comment (davidsbailey) [inline lib/assessment/label.py:147]: this was needed to fix the `413 Too Large` case, otherwise we return a 400.

File: tests/routes/conftest.py [hunk 3]
```diff
@@ -1,7 +1,155 @@
...
+def lesson_11_claude_response_body_mismatched(lesson_11_eval):
+    eval = lesson_11_eval
+    eval[0]["Key Concept"] = "Bogus Key Concept"
+    response_data = get_claude_response_data(eval)
+    yield json.dumps(response_data)
+
+@pytest.fixture
+def lesson_11_claude_response_body_too_large():
+    unparsable_json = '["unparsable JSON"'
+    response_data = get_claude_response_data(unparsable_json)
```
Comment (wilkie) [inline tests/routes/conftest.py:148]: Wasn't immediately obvious why the response data would have unparsable json until I realized that it is meant to simulate the response truncating the json. Not sure if we want some documentation of that or to update the name of this fixture, but I think it is also fine as it is.

**Review by wilkie — APPROVED** [review]
I've read through carefully and this looks very thorough. Nice! The tests pass on CI and there is some good refactoring. 👍

**Comment by davidsbailey** [comment]
Thanks for the careful review!

**Pull request merged**